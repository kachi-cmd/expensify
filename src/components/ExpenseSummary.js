import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import { Link } from 'react-router-dom';
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expense_Total'

// create a counter for expense 
// use the logic of expense total,format final figure with numeraljs
// render both to the screen ... viewing *** expenses totaling ***

export const ExpenseSummary = ({expensesCount, expensesTotal})=>{
  
  const expenseWord = expensesCount ===1 ? 'expense' : 'expenses'
  const format = numeral(expensesTotal / 100).format('$0,0.00')

    return(
    <div className="page-header">
    <div className="content-container">
     { 
expensesCount >0 && <h1 className="page-header__title"> Viewing <span>{ expensesCount }</span> {expenseWord} totaling <span>{format}</span>  </h1> 
     }
     <div className="page-header__actions">
     <Link className="button" to="/create" >Add Expense</Link>
     </div>
     </div>
    </div>
)};

const mapStateToProps = (state) => {
    const visibleExpense = selectExpenses(state.expenses, state.filters)
    return {
      expensesCount: visibleExpense.length,
      expensesTotal : selectExpensesTotal(visibleExpense)
    };
  };
  
  export default connect(mapStateToProps)(ExpenseSummary);