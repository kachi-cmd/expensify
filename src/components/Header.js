import React from 'react';
import { Link } from 'react-router-dom';
import {connect} from 'react-redux'
import {StartLogout} from '../actions/auth' 


export const Header = ({StartLogout}) => (
  <header className="header" >
  <div className="content-container">
  <div className="header__content">
    <Link className="header__title" to="/dashboard">
       <h1>The Budgeting App</h1>
    </Link>
    < button className="button button--link" onClick={StartLogout} > Logout </button>
    </div>
    </div>
  </header>
);

const mapDispatchToProps = (dispatch)=>({
  StartLogout : ()=> dispatch(StartLogout())
})

export default connect(undefined, mapDispatchToProps)(Header);
