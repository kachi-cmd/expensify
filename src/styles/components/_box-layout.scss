.box-layout{
    background: url('/images/blm.png');
    background-size: cover;
    height: 100vh;
    width: 100vw;
    display: flex;
    justify-content: center;
    align-items: center;
    
}
.box-layout__box{
  background: fade-out($color:white, $amount: .15);
  text-align: center;
  width: 25rem;
  border-radius: 3px;
  padding: $l-size $m-size; 
}
.box-layout__title{
    line-height: 1;
    margin: 0 0 $m-size 0;
}