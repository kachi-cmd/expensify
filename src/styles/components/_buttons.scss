.button{
    background:$blue;
    padding: $s-size;
    color: white;
    border: none;
    font-size: $font-size-large;
    font-weight: 300;

    display: inline-block;
    text-decoration: none;
    line-height: 1;
}
.button--link{
    background: none;
}
.remove-button{
    @extend .button;
    background: #888888;
}