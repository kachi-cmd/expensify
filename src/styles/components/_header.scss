.header{
     background: $dark-blue;
}
.header__content{
    padding: $s-size 0;
    display: flex;
    align-items: center;
    justify-content: space-between;
}
.header__title{
    color: white;
    text-decoration: none;
    h1{
        margin: 0;
    }
}