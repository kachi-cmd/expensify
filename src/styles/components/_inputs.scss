.text-input{
    height: 50px;
    font-size: $font-size-large;
    font-weight: 300 ;
    padding: $s-size;
    border: 1px solid #cacccd;
}
.select{
    @extend .text-input;
}
.textarea{
    @extend .text-input;
    height: 10rem;
}