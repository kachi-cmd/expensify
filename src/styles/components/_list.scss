.list-header{
    background: $off-white;
    border: 1px solid darken($color: $off-white, $amount: 7%);
    color: $grey;
    padding: $s-size $m-size;
    display: flex;
    justify-content: space-between;
}
.list-item{
    display: flex;
    flex-direction: column;
    border: 1px solid darken($color: $off-white, $amount: 7%);
    border-top: none;
    color: $dark-grey; 
    padding: $s-size;
    text-decoration: none;
    &:hover{
        background: $off-white;
    }
    transition: background .3s ease;
    @media (min-width: $desktop-breakpoint){
        flex-direction: row;
        justify-content: space-between;
        align-items: center;
        padding: $m-size;
    }
}
.list-item__title{
    margin: 0%;
    word-break: break-all;
}
.list-item__subtitle{
font-size: $font-size-small;
color: $grey;
}
.list-item__data{
 margin: $s-size 0 0 0;
 @media (min-width: $desktop-breakpoint){
  margin: 0%;
  padding-left: $s-size; 
}
}
.list-item--message{
    align-items: center;
    justify-content: center;
    padding: $m-size;
    color: $grey;
    &:hover{
        background: none;
    }
}
.list-body{
  margin-bottom: $m-size;
  @media (min-width: $desktop-breakpoint){
   margin-bottom: $l-size;  
}
}