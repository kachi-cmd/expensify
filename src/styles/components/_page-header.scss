.page-header{
 background: $off-white;
 padding: $l-size 0;
 margin-bottom: $l-size;
}
.page-header__title{
font-weight: 300;
 margin: 0;
 span{
     font-weight: 700;
 }
}
.page-header__actions{
    margin-top: $m-size;
}