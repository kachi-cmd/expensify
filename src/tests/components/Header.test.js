import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header';

test('should render Header correctly', () => {
  const wrapper = shallow(<Header StartLogout={() => { }} />);
  expect(wrapper).toMatchSnapshot();
});

test('should call startLogout on button click', () => {
  const StartLogout = jest.fn();
  const wrapper = shallow(<Header StartLogout={StartLogout} />);
  wrapper.find('button').simulate('click');
  expect(StartLogout).toHaveBeenCalled();
});
