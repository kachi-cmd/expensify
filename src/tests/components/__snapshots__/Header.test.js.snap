// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <[object Object]
        className="header__title"
        to="/dashboard"
      >
        <h1>
          kachi's Expensify App
        </h1>
      </[object Object]>
      <button
        className="button button--link"
        onClick={[Function]}
      >
         Logout 
      </button>
    </div>
  </div>
</header>
`;
